module/behaviour/skills/HeadBehaviourSoccer/src/HeadBehaviourSoccer.cpp

```diff
@@ -621,44 +611,44 @@ namespace behaviour {
+                                                                          SearchType sType,
+                                                                          const Sensors&,
+                                                                          const Image::Lens& lens) {
             // If there is nothing of interest, we search fot points of interest
             // log("getting search points");
             if (fixationObjects.goals.size() == 0) {
                 // log("getting search points 2");
                 // Lost searches are normalised in terms of the FOV
-                std::vector<arma::vec2> scaledResults;
+                std::vector<Eigen::Vector2d> scaledResults;
                 // scaledResults.push_back(utility::motion::kinematics::headAnglesToSeeGroundPoint(kinematicsModel,
                 // lastLocBall.position,sensors));
                 for (auto& p : searches[sType]) {
                     // log("adding search point", p.t());
                     // old angles thing
                     // Interpolate between max and min allowed angles with -1 = min and 1 = max
-                    // auto angles = arma::vec2({((max_yaw - min_yaw) * p[0] + max_yaw + min_yaw) / 2,
+                    // auto angles = Eigen::Vector2d({((max_yaw - min_yaw) * p[0] + max_yaw + min_yaw) / 2,
                     //                                    ((max_pitch - min_pitch) * p[1] + max_pitch + min_pitch) /
                     //                                    2});
 
                     // New absolute referencing
-                    arma::vec2 angles = p * M_PI / 180;
+                    Eigen::Vector2d angles = p * M_PI / 180;
                     // if(std::fabs(sensors.Htw.rotation().pitch()) < pitch_plan_threshold){
-                    // arma::vec3 lookVectorFromHead = sphericalToCartesian({1,angles[0],angles[1]});//This is an
+                    // Eigen::Vector3d lookVectorFromHead = sphericalToCartesian({1,angles[0],angles[1]});//This is an
                     // approximation relying on the robots small FOV
 
 
                     // TODO: Fix trying to look underneath and behind self!!
 
 
-                    // arma::vec3 adjustedLookVector = lookVectorFromHead;
+                    // Eigen::Vector3d adjustedLookVector = lookVectorFromHead;
                     // TODO: fix:
-                    // arma::vec3 adjustedLookVector = Rotation3D::createRotationX(sensors.Htw.rotation().pitch()) *
-                    // lookVectorFromHead;
-                    // arma::vec3 adjustedLookVector = Rotation3D::createRotationY(-pitch_plan_value) *
-                    // lookVectorFromHead;
-                    // std::vector< std::pair<ServoID, float> > goalAngles = calculateCameraLookJoints(kinematicsModel,
+                    // Eigen::Vector3d adjustedLookVector =
+                    // Eigen::Matrix3d::createRotationX(sensors.Htw.rotation().pitch())
```

Comment:
Also likely invalid

Reply:
similar story with this commented out block.